src/aotcompile.cpp

```diff
@@ -1051,6 +1051,20 @@ static AOTOutputs add_output_impl(Module &M, TargetMachine &SourceTM, ShardTimer
             // functions. We do so after optimization to avoid cloning these functions.
 
             // Float16 conversion routines
+#if defined(_CPU_X86_64_) && defined(_OS_DARWIN_) && JL_LLVM_VERSION >= 160000
```

Comment:
Can we make this dependent on the target triple? Just to preserve the fiction that codegen is cross-compile capable.